### ISSUE TITLE
Empty table row if dataSource is modified outside of user interaction

### ISSUE THREAD

**Issue by blue-lightning** [ant-design/ant-design#14500]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/p53w3x49rq)

### Steps to reproduce
The code isn't enough the reproduce the problem I'm describing, because of my limited skill.

Setup: Have an expandable table with an action column "Delete" at the end of each inner row. The data source is fetched from another location.

* Open 2 browser window, both point to the page with the table.
* Both windows: Expand 2 table rows to show their inner data. Let call these rows row1 and row2, and their inner data innerData1 and innerData2.
* Window 1: Select action "Delete" for innerData1. The action request the deletion of innerData1, fetch the new table data and trigger a re-render.
* Window 1: The table now has row1 with no expand button, while row2 is still expanded.
* Window 2: Select action "Delete" for innerData**2**. As above, the table re-render.

### What is expected?
Window 2 has both row1 and row2 with no expand button, since innerData1 was deleted from source by another client. No empty row.

### What is actually happening?
Window 2 has both row1 and row2 with no expand button, but an empty row remains in place of innerData1.

| Environment | Info |
|---|---|
| antd | 3.9.1 |
| React | 16.3.2 |
| System | Windows 10 x64 |
| Browser | Chrome 71.0 |

---
Here is an image of what happen: [Image URL](https://pasteboard.co/HXIZd9q.png). I was testing async with multiple client accesses when this happened.

**Comment by ztplz** [comment]
I guess I am a little confuse with your description and your demo did not reproduce the problem you said. Could you describe it in a clearly way?

**Comment by blue-lightning** [comment]
I guess I will have to change my code somehow. But anyway, here's the [screenshots](https://imgur.com/a/aUObz5B) demonstrating the issue for ztplz.

**Comment by zombieJ** [comment]
> I guess I will have to change my code somehow. But anyway, here's the [screenshots](https://imgur.com/a/aUObz5B) demonstrating the issue for ztplz.

Yes, please. Since screenshot can only help some style issue. But for logic related, we need code to check.
